openghg/dataobjects/_basedata.py

```diff
@@ -83,6 +83,11 @@ def __init__(
             self.data = self._zarrstore.get(version=version)
             if slice_time:
                 # If slicing by time, this must be sorted along the time dimension
+                if sort is False:
```

Comment:
Here you could add `if not sorted: ...` and maybe at the beginning we could add `sorted = np.all(self.data.time[:-1] <= self.data.time[1:])`

Reply:
I had a quick go at adding it but it's proving a little more complex as self.data isn't initially specified and there are multiple ways to add the data so data can be None initially.

Issue #1091 opened for this going forward.